content/tutorials/portenta-h7/por-openmv-gs/content.md

```diff
@@ -0,0 +1,211 @@
+
+### Pulse Width Modulation
+
+Pulse width modulation (PWM) is a way to get an artificial analog output on a digital pin. By rapidly toggling the pin from low to high. There are two parameters associated with this: the frequency of the toggling, and the duty cycle. The duty cycle is defined to be how long the pin is high compared with the length of a single period. Maximum duty cycle is when the pin is high all the time, and minimum is when it is low all the time.
+
+>On the Portenta the pins PA8, PC6, PC7, PG7, PJ11, PK1, and PH15 support PWM.
+
+### I/O Pins
+
+I/O pins are pins that can be set as input or output in the sketch. Using Micropython, there are methods to set the mode of the pin (input or output) and methods to get and set the digital logic level. We can also determine the behavior of the pull up or down resistor with the help of Micropython.
+
+## Instructions
+
+### Configuring the Development Environment
+Before you can start programming OpenMV scripts for the Portenta you need to download and install the OpenMV IDE.
+
+***IMPORTANT: Before you connect the Portenta to the OpenMV IDE make sure you update the bootloader as explained in the "Flashing the OpenMV Firmware" section!***
+
+### 1. Downloading the OpenMV IDE
+Open the [OpenMV download](https://openmv.io/pages/download) page in your browser and download the version that you need for your operating system. Alternatively, you may use the following direct download links of the OpenMV IDE 2.6.5:
+
+- [For Windows Xp, Vista, 7, 8, 10 or Later](https://github.com/openmv/openmv-ide/releases/download/v2.6.5/openmv-ide-windows-2.6.5.exe)
+- [For OsX Snow Leopard 10.6 or Later](https://github.com/openmv/openmv-ide/releases/download/v2.6.5/openmv-ide-mac-2.6.5.dmg)
+- [For Ubuntu 12.04 Lts 32-Bit or Later](https://github.com/openmv/openmv-ide/releases/download/v2.6.5/openmv-ide-linux-x86-2.6.5.run)
+- [For Ubuntu 12.04 Lts 64-Bit or Later](https://github.com/openmv/openmv-ide/releases/download/v2.6.5/openmv-ide-linux-x86_64-2.6.5.run)
+- [For Raspberry Pi 0, 1, 2, 3, and 4 or Later](https://github.com/openmv/openmv-ide/releases/download/v2.6.5/openmv-ide-linux-arm-2.6.5.tar.gz)
+
+Follow the instructions of the installer.
+
+### 2. Flashing the OpenMV Firmware
+
+Connect the Portenta to your computer via the USB-C cable if you haven't done so yet. Make sure you first update the bootloader to the latest version using the *PortentaH7_updateBootloader* sketch in the examples menu in the Arduino IDE. 
+Instructions on how to update the bootloader can be found in the ["Updating the Portenta Bootloader" tutorial](https://www.arduino.cc/pro/tutorials/portenta-h7/por-ard-bl).
+
+***In bootloader versions 17 and older there was a bug that could put the Portenta in a boot loop when the transmission aborted while flashing a large firmware file. This was fixed in the bootloader version 18. We strongly advise to update the bootloader before you proceed with the next step of this tutorial!***
```

Comment:
I would add this as a troubleshooting or move it to he bootloader tutorial itself